core/processor/src/main/java/io/quarkus/annotation/processor/documentation/config/formatter/JavadocToMarkdownTransformer.java

```diff
@@ -0,0 +1,87 @@
+package io.quarkus.annotation.processor.documentation.config.formatter;
+
+import java.util.regex.Pattern;
+
+import com.github.javaparser.StaticJavaParser;
+import com.github.javaparser.javadoc.Javadoc;
+import com.github.javaparser.javadoc.description.JavadocDescription;
+import com.github.javaparser.javadoc.description.JavadocDescriptionElement;
+import com.github.javaparser.javadoc.description.JavadocInlineTag;
+
+import io.quarkus.annotation.processor.documentation.config.model.JavadocFormat;
+
+public class JavadocToMarkdownTransformer {
+
+    private static final Pattern START_OF_LINE = Pattern.compile("^", Pattern.MULTILINE);
+
+    public static String toMarkdown(String javadoc, JavadocFormat format) {
+        if (javadoc == null || javadoc.isBlank()) {
+            return null;
+        }
+
+        if (format == JavadocFormat.MARKDOWN) {
+            return javadoc;
+        } else if (format == JavadocFormat.JAVADOC) {
+            // the parser expects all the lines to start with "* "
+            // we add it as it has been previously removed
+            Javadoc parsedJavadoc = StaticJavaParser.parseJavadoc(START_OF_LINE.matcher(javadoc).replaceAll("* "));
+
+            // HTML is valid Javadoc but we need to drop the Javadoc tags e.g. {@link ...}
+            return simplifyJavadoc(parsedJavadoc.getDescription());
+        }
+
+        // it's Asciidoc, the fun begins...
+        return "";
```

Comment:
Yeah, that's the plan.

I don't expect it to be perfect as I won't implement something heavy (I would like to avoid including AsciiDoctorJ + JRuby if we can) but hope it's going to be good enough for the IDEs to be able to consume it.
And we will improve over time when edge cases pop up.

If you have the Javadoc edge cases still around, I'm interested in making it better (if possible). It will never be perfect but it would be nice for it to be good enough in most cases.

But in any case, AsciiDoc will be supported if that's what worried you.

Reply:
The IDE work is slowly being put in shape here: https://github.com/quarkusio/quarkus/pull/42677 .

I need to rebase and wire the Markdown transformer and... then... implement it somehow :).

Reply:
There's this thing in Javascript that looks interesting and matches what I would like to achieve: https://github.com/opendevise/downdoc .

But I'm not sure how easy it would be to convert it in Java. The code is not exactly readable :).

Not sure if AI could help translating downdoc to Java.